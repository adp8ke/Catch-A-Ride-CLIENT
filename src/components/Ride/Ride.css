.ride-info {
  font-size: 12px;
  border: 1px solid black;
  top: 0px;
  margin-top: 0;
  padding: 5px 5px;
  padding-top: 5px;
  padding-bottom: 5px;
  text-align: left;
  background: #CBEAEF;
  border: #CBEAEF 1px solid;
  border-radius: 5px 5px 0 0;
  color: black;
}

.labels {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.locations {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}
