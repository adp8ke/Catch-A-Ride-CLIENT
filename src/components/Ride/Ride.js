import React, { Component } from 'react';

import moment from 'moment';
import './Ride.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMapMarkerAlt } from '@fortawesome/free-solid-svg-icons';

class Ride extends Component {
  render() {
    const date_time = moment(this.props.ride.date_time).format('MM/DD/YYYY @ h:mmA');
    let remainingSeats = 0;
    let count = 0;

    let sRArray = Object.keys(this.props.ride);

    for(let i = 6; i < sRArray.length; i++) {
      count++;
      if(this.props.ride.capacity < count) {
        break;
      }

      if(this.props.ride[sRArray[i]] === null) {
        remainingSeats++;
      }
    }
    if(remainingSeats === 0) {
      remainingSeats = 'This ride is full';
    }

    return (
      <>
        <p className='ride-info'>
          {date_time}
        </p>
        
        <span className='labels'>
          <h3 className='locations'>Meetup<br />Location</h3>
          <h3 className='locations'>Destination</h3>
        </span>
        <span className='locations'>
          <p className='location'><FontAwesomeIcon icon={faMapMarkerAlt}/> {this.props.ride.starting}</p>
          <p className='destination'><FontAwesomeIcon icon={faMapMarkerAlt}/> {this.props.ride.destination}</p>
        </span>
        <p className='driver'>Driver Name: {this.props.ride.driver_name}</p>
        <p>Capacity: {this.props.ride.capacity}</p>
        <p>Remaining Seats: {remainingSeats}</p>
        <p>Description: {this.props.ride.description}</p>
      </>
    );
  }
}

export default Ride;
