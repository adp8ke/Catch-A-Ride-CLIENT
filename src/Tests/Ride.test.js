import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter} from 'react-router-dom';
import Ride from '../components/Ride/Ride';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const ride = {
    id: '154a8af7-4f6a-4c65-ab44-e8f659cf6797',
    driver_id: '113938795112280626788',
    starting: 'Virginia Beach',
    destination: 'Blacksburg',
    description: '$10 for gas, limited trunk space',
    driver_name: 'Andrew Yin',
    date_time: '2019-09-17T19:00:00.000Z',
    capacity: 4,
    p1: '101820519124146512532',
    p2: null,
    p3: null,
    p4: null,
    p5: null,
    p6: null,
    p7: null
  };
  ReactDOM.render(<MemoryRouter><Ride ride={ride} /></MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});
