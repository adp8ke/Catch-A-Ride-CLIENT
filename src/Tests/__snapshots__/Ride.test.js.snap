// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <p
    className="ride-info"
  >
    09/17/2019 @ 2:00PM
  </p>,
  <span
    className="labels"
  >
    <h3
      className="locations"
    >
      Meetup
      <br />
      Location
    </h3>
    <h3
      className="locations"
    >
      Destination
    </h3>
  </span>,
  <span
    className="locations"
  >
    <p
      className="location"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-map-marker-alt fa-w-12 "
        data-icon="map-marker-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
      Virginia Beach
    </p>
    <p
      className="destination"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-map-marker-alt fa-w-12 "
        data-icon="map-marker-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
      Blacksburg
    </p>
  </span>,
  <p
    className="driver"
  >
    Driver Name: 
    Andrew Yin
  </p>,
  <p>
    Capacity: 
    4
  </p>,
  <p>
    Remaining Seats: 
    1
  </p>,
  <p>
    Description: 
    $10 for gas, limited trunk space
  </p>,
]
`;
